tests/unit/nupic/algorithms/sdr_classifier_test.py

```diff
@@ -0,0 +1,445 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""Unit tests for SDRClassifier module."""
+
+CL_VERBOSITY = 0
+
+import cPickle as pickle
+import types
+import unittest2 as unittest
+
+import numpy
+import tempfile
+
+from nupic.algorithms.sdr_classifier import SDRClassifier
+
+class SDRClassifierTest(unittest.TestCase):
+  """Unit tests for SDRClassifier class."""
+
+
+  def setUp(self):
+    self._classifier = SDRClassifier
+
+
+  def testInitialization(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    self.assertEqual(type(c), self._classifier)
+
+
+  def testSingleValue(self):
+    """Send same value 10 times and expect high likelihood for prediction."""
+    classifier = self._classifier(steps=[1], alpha=1.0)
+
+    # Enough times to perform Inference and learn associations
+    retval = []
+    for recordNum in xrange(10):
+      retval = self._compute(classifier, recordNum, [1, 5], 0, 10)
+
+    self.assertEqual(retval['actualValues'][0], 10)
+    self.assertGreater(retval[1][0], 0.9)
+
+
+  def testSingleValue0Steps(self):
```

Comment:
Can we have a test with steps=[0] that distinguishes between steps=[1]? If we alternate the classification between two buckets, 0-step should learn one of them and 1-step should learn the other.